PIL/Image.py

```diff
@@ -600,6 +636,7 @@ def _dump(self, file=None, format=None, **options):
         return file
 
     def __eq__(self, other):
+        # type: (object) -> bool
         return (self.__class__.__name__ == other.__class__.__name__ and
                 self.mode == other.mode and
```

Comment:
We're getting errors here now, that object doesn't have all of these attributes, and we're getting that because the type system isn't picking up that we've narrowed the type of other to an Image with the first comparison.

Reply:
The class checking was added here: https://github.com/python-pillow/Pillow/commit/1aee9bfdef1d73007d8e35ab883c8c936c1da602  I'm not clear if we're checking class names here because an isInstance would be a performance hit here or if this is just a complicated way of doing the class check.

Reply:
I see you merged this approach in my other PR?

The difference is maybe that it doesn't do an exact class match, but rather based on the base class? So derived classes will match with `isinstance`. Depending on the perceived slowdown/importance, as long as there is an isinstance check above the tests which use Image attributes, it should be fine (and mypy shouldn't then complain).